Training/Training/Training.c

```diff
@@ -70,32 +67,31 @@ void BufferDrain () {
 
 /// <summary>Function to check if the given input is a phrase or a number</summary>
 bool IsValidNum (char* phrase, int* num) {
-   char* endptr;
+   errno = 0;
+   char* endptr = NULL;
    *num = strtol (phrase, &endptr, 10);
-   if (endptr == phrase || *endptr != '\0' || errno == ERANGE || phrase[0] == ' ' || phrase[0] == '\t')
-      return false;
-   else
-      return true;
+   return endptr == phrase || *endptr != '\0' || errno == ERANGE || phrase[0] == ' ' || phrase[0] == '\t' ?
+      false : true;
 }
 
 /// <summary>Function to run test cases</summary>
 void TestCases () {
-   int Length, pChecker, num;
-   long long int rev;
+   int Length;
    char* Inputs[] = { "Was it a car or a cat I saw?", "I did, did I?", "Don't nod", "TRUMPF",
-      "Metamation", "121", "232", "123454321", "123456789", "0", "!@#  $%", " " }, * Expected[] = { "Palindrome", "Palindrome",
-      "Palindrome", "Not a Palindrome", "Not a Palindrome","Palindrome", "Palindrome",
-      "Palindrome", "Not a Palindrome", "Palindrome", "Invalid", "Invalid" }, * output = NULL;
+      "Metamation", "121", "232", "123454321", "123456789", "0", "!@#  $%", " ", "-878" }, * Expected[] = {
+      "Palindrome", "Palindrome", "Palindrome", "Not a Palindrome", "Not a Palindrome","Palindrome",
+      "Palindrome", "Palindrome", "Not a Palindrome", "Palindrome", "Invalid", "Invalid", "Not a Palindrome" },
+      * output = NULL;
    Length = sizeof (Inputs) / sizeof (Inputs[0]);
    printf (ANSI_COLOR_YELLOW"\n-------------------------------------------------------------------"
            "--------------\n|------------Input-------------|-----Expected-----|------Output------"
            "|--Result--|"ANSI_COLOR_RESET);
    for (int i = 0; i < Length; i++) {
-      rev = 0;
-      if (IsValidNum (Inputs[i], &num))
-         pChecker = NumberReverser (num, &rev);
-      else
-         pChecker = PalindromeChecker (Inputs[i]);
+      int num, pChecker;
+      long long int rev = 0;
+      pChecker = IsValidNum (Inputs[i], &num) ?
+         ((NumberReverser (num) == num && num >= 0) ? PALINDROME : NOTPALINDROME) :

```

Comment:
Here you can check whether the num >=0 and then call the NumberReverser function, this can provide minor optimisation in case of negative numbers and can avoid reversing the number and directly return NOT PALINDROME.